api/src/main/proto/stellarstation/api/v1/radio/radio.proto

```diff
@@ -89,4 +89,153 @@ enum Modulation {
+  PSK16 = 6;
+
+  // 32-Phase Shift Keying. https://en.wikipedia.org/wiki/Phase-shift_keying
+  PSK32 = 7;
+
+  // 64-Phase Shift Keying. https://en.wikipedia.org/wiki/Phase-shift_keying
+  PSK64 = 8;
+
+  // 128-Phase Shift Keying. https://en.wikipedia.org/wiki/Phase-shift_keying
+  PSK128 = 9;
+
+  // 256-Phase Shift Keying. https://en.wikipedia.org/wiki/Phase-shift_keying
+  PSK256 = 10;
+
+
+  // Offset Quadrature Phase Shift Keying. https://en.wikipedia.org/wiki/Phase-shift_keying#Offset_QPSK_(OQPSK)
+  OQPSK = 11;
+
+
+  // 8-Phase Amplitude Modulation. https://en.wikipedia.org/wiki/Quadrature_amplitude_modulation
+  QAM8 = 12;
+
+  // 16-Phase Amplitude Modulation. https://en.wikipedia.org/wiki/Quadrature_amplitude_modulation
+  QAM16 = 13;
+
+  // 32-Phase Amplitude Modulation. https://en.wikipedia.org/wiki/Quadrature_amplitude_modulation
+  QAM32 = 14;
+
+  // 64-Phase Amplitude Modulation. https://en.wikipedia.org/wiki/Quadrature_amplitude_modulation
+  QAM64 = 15;
+
+  // 128-Phase Amplitude Modulation. https://en.wikipedia.org/wiki/Quadrature_amplitude_modulation
+  QAM128 = 16;
+
+  // 256-Phase Amplitude Modulation. https://en.wikipedia.org/wiki/Quadrature_amplitude_modulation
+  QAM256 = 17;
+
+
+  // Minimum Shift Keying. https://en.wikipedia.org/wiki/Minimum-shift_keying
+  MSK = 18;
+
+  // Gaussian Minimum Shift Keying. https://en.wikipedia.org/wiki/Minimum-shift_keying#Gaussian_minimum-shift_keying
+  GMSK = 19;
+
+
+  // Amplitude modulation (generic)
+  AM = 20;
+
+  // Frequency modulation (generic)
+  FM = 21;
+
+  // Phase modulation (generic)
+  PM = 22;
+}
+
+// Line coding applied during modulation. Line coding is the technique for modeling binary 0 and 1
+// within a modulated waveform.
+enum LineCoding {
+  // Non-return-to-zero level.
+  // https://en.wikipedia.org/wiki/Non-return-to-zero
+  NRZ_L = 0;
+
+  // Non-return-to-zero mark (i.e., differential encoding).
+  // https://en.wikipedia.org/wiki/Non-return-to-zero
+  NRZ_M = 1;
+
+  // Non-return-to-zero space (i.e., non-differential-encoding).
+  // https://en.wikipedia.org/wiki/Non-return-to-zero
+  NRZ_S = 2;
+
+  // Return-to-zero. https://en.wikipedia.org/wiki/Return-to-zero
+  RZ = 3;
+
+  // Bipolar return-to-zero level.
+  // https://en.wikipedia.org/wiki/Bipolar_encoding
+  BP_L = 4;
+
+  // Bipolar return-to-zero mark (i.e., differential encoding).
+  // https://en.wikipedia.org/wiki/Bipolar_encoding
+  BP_M = 5;
+
+  // Bipolar return-to-zero space (i.e., non-differential encoding).
+  // https://en.wikipedia.org/wiki/Bipolar_encoding
+  BP_S = 6;
+}
+
+// Parameters for convolutional coding applied to a bitstream.
+// https://en.wikipedia.org/wiki/Convolutional_code
+message ConvolutionalCodingParams {
+  // The constraint length, `k`, of the code.
+  uint32 k = 1;
+
+  // The inverse of the rate of coding, e.g., if rate is `1 / r`, `inverse_rate` is `r`.
+  uint32 inverse_rate = 2;
+
+  message DescramblingParams {
```

Comment:
Sorry, should've been clearer. Descrambling params should be on the same level as convolutional coding params since they're not related.

convolutional coding params:
 - k
 - inverse rate
 - polynomials (`repeated bytes` where each of the r `bytes` has length 'k' bits)

descrambling params:
 - type (additive/multiplicative)
 - length
 - polynomial mask for shift register (single `bytes` of length `length` bits)
 - seed (single `bytes` of length `length` bits)

Reply:
Thanks - sorry for not knowing anything about this stuff ;)